src/EventHandlers.ts

```diff
@@ -0,0 +1,185 @@
+import {
+  SweepStakesNFTsContract_Transfer_loader,
+  SweepStakesNFTsContract_Transfer_handler,
+  SweepStakesNFTsContract_Enter_loader,
+  SweepStakesNFTsContract_Enter_handler,
+  SweepStakesNFTsContract_WinnerAssigned_loader,
+  SweepStakesNFTsContract_WinnerAssigned_handler,
+  SweepStakesNFTsContract_Unstake_loader,
+  SweepStakesNFTsContract_Unstake_handler,
+} from "../generated/src/Handlers.gen";
+import { sweepStakesNFTs_WinnerAssignedEntity } from "./src/Types.gen";
+
+// TODO: move to a Constants file.
+const CONTRACT_NAME = 'SweepStakes'
+
+SweepStakesNFTsContract_Transfer_loader(({ event, context }) => {
+  const tokenId = event.params.tokenId.toString();
+  context.Token.load(tokenId);
+});
+
+SweepStakesNFTsContract_Transfer_handler(({ event, context }) => {
+  const tokenId = event.params.tokenId.toString();
+  const newOwner = event.params.to;
+  const tokenEntity = context.Token.get(tokenId);
+
+  const sweepStakesNFTs_TransferEntity = {
+    id: event.transactionHash + event.logIndex.toString(),
+    from: event.params.from,
+    to: event.params.to,
+    tokenId: event.params.tokenId,
+  };
+
+  if (tokenEntity !== undefined) {
+    context.Token.set({
+      ...tokenEntity, userAddress: newOwner,
+    });
+  } else {
+    context.Token.set({
+      id: tokenId,
+      userAddress: newOwner,
+      totalStakes: 0n,
+      totalPrizes: 0n,
+      balance: 0n,
+    });
+  }
+
+  context.SweepStakesNFTs_Transfer.set(sweepStakesNFTs_TransferEntity);
+});
+
+SweepStakesNFTsContract_Enter_loader(({ event, context }) => {
+  const tokenId = event.params._tokenId.toString();
+  context.Token.load(tokenId);
+  context.ContractTotals.load(CONTRACT_NAME);
+});
+
+SweepStakesNFTsContract_Enter_handler(({ event, context }) => {
+  const tokenId = event.params._tokenId.toString();
+  const tokenEntity = context.Token.get(tokenId);
+  if (tokenEntity === undefined) {
+    throw new Error('Token not found, error in handler. It should have been created by the "Transfer" event handler.')
+  }
```

Comment:
I put these errors in here - since they should always be defined.

Basically there could be a bug where you forget to set the tokenEntity in the first transfer and then you'd hit this error. Basically if it ever isn't in the database.